### PR TITLE
Working basic denoising conditional CVAE on MNIST data

### PR DESCRIPTION
**Pull request by bamford**
This is based on the example Larissa found together with both my earlier intuitions and some further thinking about where to include the conditions in the network. Again, decent examples seem to be hard to find online. Maybe I should write a Medium article about the steps we are taking in developing this code!

I've extended the previous CVAE example to add in a condition: the label of the digit. I have included some comments in the code explaining my thought process, and how this MNIST example relates to our real goals (i.e. conditioning on a continuous quantity, redshift, rather than categories).

This is my first go at this, but it seems to work (based on very limited testing!). After just 10 epochs of training (loss still falling, should actually train for longer), with just 2 latent dimensions and beta=1, we get `loss: 32.9153 - reconstruction_loss: 29.9338 - kl_loss: 2.9880 - val_loss: 32.9985 - val_reconstruction_loss: 30.0675 - val_kl_loss: 2.9521` and [pretty good looking reconstructions](https://github.com/aidenrolfe/ARG/files/6058379/examples_latent2_beta1_epochs10.pdf).

### CODE REVIEW

**Comment by bamford** [comment]
Here's training with the settings used in my previous pull request (#4), 5 latent dimensions and beta=0.1, training for 60 epochs, which is pretty much converged... `loss: 28.1133 - reconstruction_loss: 27.5096 - kl_loss: 0.6034 - val_loss: 29.3006 - val_reconstruction_loss: 28.7094 - val_kl_loss: 0.5971`. Final losses aren't quite as low, but the [reconstructions are great](https://github.com/aidenrolfe/ARG/files/6060036/examples.pdf)!

**Comment by larissapalethorpe** [comment]
Just as a follow up I ran this with 5 latent dimensions and a beta of 0.1 over 100 epochs which took about 2 and half hours to run, which gave  ` loss: 27.6345 - reconstruction_loss: 27.0245 - kl_loss: 0.6092 - val_loss: 29.1109 - val_reconstruction_loss: 28.5050 - val_kl_loss: 0.5988` but again gave [pretty good constructions!](url
[MNIST_CVAE2.pdf](https://github.com/aidenrolfe/ARG/files/6064237/MNIST_CVAE2.pdf)
)

**Comment by bamford** [comment]
Looking good. Feel free to experiment and make plots.
